### PR TITLE
Checkout page progress

### PR DESCRIPTION
**Pull request by AlanPieczonka**
Basic logic, MINIMAL product card and payment UI (I guess we can polish that up later), for now I also went with just creating normal instances of the modal class + custom hook instead of creating reusable modal component but let me know what you think about it. I also used setTimeout instead of doing an actual request when submitting order because from what I see on backend we need userId there and we don't have it yet. 
I could describe every single change but I guess it'll be easier if you ask about specific thing when reviewing the PR and having objections/questions etc. 
Cheers :)

### CODE REVIEW

File: src/reducers/order.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          quantity: product.quantity + 1 <= 10 ? product.quantity + 1 : 10
+         } : product
+      })
+    case DECREASE_PRODUCT_QUANTITY:
+      return state.map((product, index) => {
+        return index === action.payload.index ? { 
+          ...product,
+          quantity: product.quantity - 1 >= 1 ? product.quantity - 1 : 1
+         } : product
+      })
```
Comment (Drakeoon) [inline src/reducers/order.js]: This will get cleaned up if we decide to go with UPDATE_PRODUCT_QUANTITY, consider it

File: src/utils/order.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export const calculateSum = (order) => order.length
+? order
+    .map(({ price, quantity }) => price * quantity)
+    .reduce((x, y) => x + y)
+    .toFixed(2)
```
Comment (Drakeoon) [inline src/utils/order.js]: Consider using lodash and this conditional approach to refactor this a bit:
```suggestion
if (!order.length) {
 return 0
}

return sumBy(order, ({ price, quantity }) => price * quantity)
```

File: src/actionCreators/order.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import * as types from "../types";
+
+export const addProduct = payload => ({ type: types.ADD_PRODUCT, payload })
+export const removeProduct = payload => ({ type: types.REMOVE_PRODUCT, payload })
+export const decreaseProductQuantity = payload => ({ type: types.DECREASE_PRODUCT_QUANTITY, payload })
+export const increaseProductQuantity = payload => ({ type: types.INCREASE_PRODUCT_QUANTITY, payload })
```
[the remark this answers is not in the document]
Comment (AlanPieczonka) [reply src/actionCreators/order.js]: Yes I agree, I'll name removeProduct's argument an "index" becasue that's what we actually pass there, same as in updateProductQuantity, and I'll change addProduct's argument to product as suggested.

File: src/assets/styles/components/button.sass [hunk 4]
```diff
@@ -22,6 +22,10 @@
   &.-block
     width: 100%
 
+  &.-grayed
+    border-color: $color-gray-80
+    background-color: $color-gray-80
+
```
[the remark this answers is not in the document]
Comment (AlanPieczonka) [reply src/assets/styles/components/button.sass]: Yeah that will be better 👍👍👍

File: src/reducers/order.js [hunk 5]
```diff
@@ -13,20 +13,13 @@ export default function(state = initialState, action) {
...
       })
-    case INCREASE_PRODUCT_QUANTITY:
+    case UPDATE_PRODUCT_QUANTITY:
       return state.map((product, index) => {
         return index === action.payload.index ? { 
           ...product,
-          quantity: product.quantity + 1 <= 10 ? product.quantity + 1 : 10
+          quantity: action.payload.quantity
          } : product
-      })
```
Comment (Drakeoon) [inline src/reducers/order.js]: Forgot to add it in my previous review, but we can refactor this a little. Consider applying this approach:
```javascript
const orders = [...]

const newOrders = [...orders]
newOrders[index].quantity = action.payload.quantity

return newOrders
```

**Pull request merged**